include/Kernel.h

```diff
@@ -91,9 +91,19 @@ class Kernel
    *  the linear solve
    */
   virtual void execute(
-    SharedMemView<DoubleType**>&,
-    SharedMemView<DoubleType*>&,
-    ScratchViews<DoubleType>&)
+    SharedMemView<DoubleType**> &lhs,
```

Comment:
Wouldn't this create unused variable warnings?

Reply:
I wouldn't think that should create unused variable warnings. If it does, I guess we could either comment out the argument, e.g., /*lhs*/ or remove the curly braces to make the method pure virtual. In general I find it annoying to see prototypes with only types but no argument names.

Reply:
I thought that the header can have all of the variable names while if an implementation does not need the variable, one would:

SharedMemView<DoubleType**> &/*lhs*/

I do not prefer the bare signatures, however, am fine with whomever is writing the class/kernel to decide:)

Reply:
Yeah, Alan had provided empty implementations in the header (so method definition and not declaration), so I thought it would issue warnings. However, it seems that Clang on OSX  doesn't complain about it. Sorry for the noise.